### ISSUE TITLE
New hook proposal: useTitleTemplate

### ISSUE THREAD

**Issue by gustavpursche** [JoviDeCroock/hooked-head#4]
Hi,
after your little lib popped up last week I had a chance to have a look at it today: nice work :rocket:

One thing I really like(d) about `react-helmet`, because it's a very common usecase, is the `titleTemplate` prop

```jsx
<Helmet titleTemplate="%s | MyAwesomeWebsite">
  <title>Nested Title</title>
</Helmet>
```

It makes updating nested titles very easy. Of course this could easily be achieved by wrapping `useTitle` into it's custom local hook.

What do you think about either:

1. adding a `useTitleTemplate(title: string)` hook. All following `useTitle` calls would then replace `%s` with the title

2. extending `useTitle` to accept a second parameter: `useTitle(title: string, { template: '%s | MyAwesomeWebsite' } : { template: string })`

I'm happy to work on this - just wanted to see first, what you'd think about the idea.

**Comment by JoviDeCroock** [comment]
Hey,

Thank you so much for the kind words, I think option 1 should be the easiest to achieve, this hook would need to register itself to the `dispatcher` since well most commonly there won't be a lot of these so I don't think it's worth implementing complex queueing logic for this.

So it registers itself in a useEffect and then gets used when `process` runs to use this template.

Does this sound reasonable to you?

**Comment by gustavpursche** [comment]
It does. Thanks for the hints - I'll have a look!

**Comment by gustavpursche** [comment]
In case someone discovers this issue in the future: find the PR here: https://github.com/JoviDeCroock/hooked-head/pull/6

**Comment by JoviDeCroock** [comment]
This has been released and will be part of the 0.3.0 release!

**Comment by JoviDeCroock** [comment]
@gustavpursche here is your feature in action, https://9joqi.codesandbox.io/ 🎉 thanks again!
